crypto/x509/x509_lu.c

```diff
@@ -670,36 +677,28 @@ STACK_OF(X509) *X509_STORE_CTX_get1_certs(X509_STORE_CTX *ctx,
          * Nothing found in cache: do lookup to possibly add new objects to
          * cache
          */
-        X509_OBJECT *xobj = X509_OBJECT_new();
-
         X509_STORE_unlock(store);
-        if (xobj == NULL)
-            return NULL;
-        i = ossl_x509_store_ctx_get_by_subject(ctx, X509_LU_X509, nm, xobj);
-        if (i <= 0) {
-            X509_OBJECT_free(xobj);
+        i = ossl_x509_store_ctx_get_by_subject(ctx, X509_LU_X509, nm, NULL);
```

Comment:
Some day, it might be good to not assume that cached data is *always* complete and correct. Trust anchors can become untrusted, and new ones can come along. Certainly with CApath, it makes some sense to periodically ignore the cached data, and do a fresh lookup, perhaps a new control can set the refresh interval. Nothing needs to be done in this PR, but something to keep in mind...

Reply:
Good point.
I suggest opening an issue to help keeping this in mind.